### PR TITLE
perf(artistlists): boost performance of big partner lists

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [PHIRE-601] 

### Description

Boosts a bit the performance (around 26%) for bigger partners.

Used [this guide ](https://reactnative.dev/docs/optimizing-flatlist-configuration) from rn docs and [this perf monitoring](https://github.com/bamlab/react-native-flipper-performance-monitor) tool that I just added in energy.

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[PHIRE-601]: https://artsyproduct.atlassian.net/browse/PHIRE-601?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/system/sync/artifacts/__generatedRelayChecksum.ts [hunk 1]
```diff
@@ -5,4 +5,4 @@
  * also part of the `syncManager` file, and any dependency they might have (eg. fragments).
  */
 
-export const relayChecksum = "7715d44c0a995e88c1b6381c6f115a6277c32c13" // pragma: allowlist secret
+export const relayChecksum = "4fa53981da69c6cd879c42270b14a465838a5aa1" // pragma: allowlist secret
```
Comment (gkartalis) [inline src/system/sync/artifacts/__generatedRelayChecksum.ts:8]: 🤔 is this supposed to get refreshed?

File: src/components/Lists/ArtistsList.tsx [hunk 2]
```diff
@@ -79,15 +94,13 @@ export const artistsListQuery = graphql`
   query ArtistsListQuery($partnerID: String!) {
     partner(id: $partnerID) {
       allArtistsConnection(includeAllFields: true) {
-        totalCount
```
Comment (gkartalis) [inline src/components/Lists/ArtistsList.tsx:82]: removed those since we didn't use them anywhere

File: src/components/Lists/ArtistsList.tsx [hunk 3]
```diff
@@ -48,26 +75,14 @@ export const ArtistsList: React.FC<ArtistsListProps> = ({
     <ArtistsFlatList
       data={items}
       numColumns={1}
-      initialNumToRender={30}
+      maxToRenderPerBatch={20}
+      initialNumToRender={20}
```
Comment (gkartalis) [inline src/components/Lists/ArtistsList.tsx:79]: https://reactnative.dev/docs/optimizing-flatlist-configuration#maxtorenderperbatch

File: src/system/sync/artifacts/__generatedRelayChecksum.ts (hunk 1 above)
Comment (damassi) [reply src/system/sync/artifacts/__generatedRelayChecksum.ts:8]: Yup, when relay changes happen, we checksum things to show that a new sync is available.

**Pull request merged**